### ISSUE TITLE
Have `InterfaceCaseProvider.case_runners` assign case data to `wrapper` attribute

### ISSUE THREAD

**Issue by rtweeks** [PayTrace/intercom_test#15]
It can be useful to the testing framework consuming this library to have access to the case data outside the scope of the test function.  One example would be including information from the case (HTTP method and path, or some kind of "story") in the verbose name reported for the test.

This could be as simple as:
```python
class InterfaceCaseProvider:
    ...
    def case_runners(self, fn, *, do_compact_updates=True):
        ...

        # new code
        wrapper.case = case

        # existing code
        yield wrapper
```

**Comment by rtweeks** [comment]
I'm working on a patch for this.

**Comment by rtweeks** [comment]
Fixed in v2.3.0
